fenix/app/src/main/java/org/mozilla/fenix/home/intent/ReEngagementIntentProcessor.kt

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.intent
+
+import android.content.Intent
+import androidx.navigation.NavController
+import androidx.navigation.navOptions
+import mozilla.components.concept.engine.EngineSession
+import mozilla.telemetry.glean.private.NoExtras
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.GleanMetrics.Events
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.NavGraphDirections
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.onboarding.ReEngagementNotificationWorker
+import org.mozilla.fenix.onboarding.ReEngagementNotificationWorker.Companion.isReEngagementNotificationIntent
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Handle when the re-engagement notification is tapped
+ *
+ * This should only happens once in a user's lifetime notification,
+ * [settings.shouldShowReEngagementNotification] will return false if the user already seen the
+ * notification.
+ */
+class ReEngagementIntentProcessor(
+    private val activity: HomeActivity,
+    private val settings: Settings,
```

Comment:
❓ Could we pass in the least we need, just the `reEngagementNotificationType` Integer? Makes it easier to reason about looking at the class that it just reads that and doesn't use settings in any other way.

Reply:
I don't think we can pass in the integer as a parameter since the value could change depending when we receive the Nimbus experiment value.  (Technically this should be a first run experiment but we can't assume that).  I can pass in a () -> Int but not sure if that's better.

Reply:
After some thoughts, I think I'll still pass in settings.  Tried passing () -> Int made the code a bit harder to read.

Reply:
I see, thanks for the explanation! 👍